framework/src/main/java/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -865,6 +865,36 @@ protected void checkExtendsOrImplements(
             AnnotationMirrorSet classBounds,
             TypeMirror classType,
             boolean isExtends) {
+        checkExtendsOrImplements(boundClause, classBounds, classType, isExtends, true);
```

Comment:
Instead of adding another overload with another boolean, have you considered splitting out the extra check into a helper method? And then only that helper method needs to be overridden or adapted in checkers that allow annotations?